File: ruby/ql/src/queries/security/cwe-020/examples/IncompleteHostnameRegExp.rb
class AppController < ApplicationController

    def index
        url = params[:url]
        host = URI(url).host
        # BAD: the host of `url` may be controlled by an attacker
        regex = /^((www|beta).)?example.com/
        if host.match(regex)
            redirect_to url
        end
    end

end

File: javascript/ql/src/Security/CWE-020/examples/IncompleteHostnameRegExp.js
app.get('/some/path', function(req, res) {
    let url = req.param('url'),
        host = urlLib.parse(url).host;
    // BAD: the host of `url` may be controlled by an attacker
    let regex = /^((www|beta).)?example.com/;
    if (host.match(regex)) {
        res.redirect(url);
    }
});
